module-2/backend/week_three.md

```diff
@@ -9,15 +9,20 @@ Note: When you're done, submit a PR with a reflection in the comments about how
 
 ### Questions
 
-1. What is the entry at the command line to create a new rails app?
-2. What do Models generally inherit from in rails?
-3. What do Controllers generally inherit from in a rails project?
-4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality?
-5. What rake task is useful when looking at routes, and what information does it give you?
-6. What is an example of a route helper? When would you use them?
-7. What's the difference between what `_url` and `_path` return when combined with a routes prefix?
-8. What are strong params and why are the necessary?
-9. What role does `form_for` play in helping us create our forms?
-10. How does `form_for` know where to submit the user's input?
+1. What is the entry at the command line to create a new rails app? rails new name -T --database=postgresql --skip-spring --skip-turbolinks
+2. What do Models generally inherit from in rails? ApplicationRecord
+3. What do Controllers generally inherit from in a rails project? ApplicationController
+4. How would I create a route if I wanted to see a specific horse in my routes fitle assuming I'm sticking to standard conventions and that I didn't want other CRUD functionality? resources :horses, only: [:show]
+5. What rake task is useful when looking at routes, and what information does it give you? rake routes, For each route it tells you the path helper, uri, and controller and action.
+6. What is an example of a route helper? When would you use them? Example: companies_path, You would use them with link_to or in a capybara test.
+7. What's the difference between what `_url` and `_path` return when combined with a routes prefix? I had to look this one up. _url will return an absolute path while _path is a path relative to the route. If you need to use the link outside of your own site, you would use _url.
+8. What are strong params and why are the necessary? Strong params are a private method that verifies that the params generated when creating a new thing actually came from submitting a form. It prevents a user from generating params by typing them in manually.
+9. What role does `form_for` play in helping us create our forms? It is the command that rails used to create a form and gather the params from that form.
+10. How does `form_for` know where to submit the user's input? Rails magic??? I think it has something to do with calling thing.new in the create method. In the update method values already exist so it knows to go to update instead of create.
```

Comment:
the instance variables that we are using are providing a model/models and path helper on where to submit